src/client/client.go

```diff
@@ -23,11 +23,7 @@ type Kubernetes interface {
 	// FindNode returns a Node reference containing the pod named as the argument, if any
 	FindNode(name string) (*v1.Node, error)
 	// FindPodsByLabel returns a PodList reference containing the pods matching the provided name/value label pair
-	FindPodsByLabel(name, value string) (*v1.PodList, error)
-	// FindPodByName returns a PodList reference that should contain the pod whose name matches with the name argument
-	FindPodByName(name string) (*v1.PodList, error)
-	// FindPodsByHostname returns a Podlist reference containing the pod or pods whose hostname matches the argument
-	FindPodsByHostname(hostname string) (*v1.PodList, error)
+	FindPodsByLabel(namespace, labelSelector string) (*v1.PodList, error)
```

Comment:
I think it might be good to put an example in this comment regarding how `labelSelector` is supposed to look, just in case someone unfamiliar with the k8s concept reads this.

Reply:
I could actually make it strongly typed and use https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector. What do you think?

Reply:
I think it would be quite nice. How would the query behave if supplied with the zero value of the `LabelSelector`?

Reply:
I'll make sure it selects all pods :)

Reply:
Then I'd say let's go with the strong type

Reply:
Done in #172